s3tests_boto3/functional/test_s3select.py

```diff
@@ -1162,3 +1196,35 @@ def test_bool_cast_expressions():
     res_s3select = remove_xml_tags_from_result(  run_s3select(bucket_name,csv_obj_name, 'select count(*) from s3object where cast(_1 as int) != 0 ;')).replace("\n","")
 
     s3select_assert_result( res_s3select_cast, res_s3select )
+
+@attr('s3select')
+def test_progress_expressions():
+
+    csv_obj = create_random_csv_object(1000000,10)
+
+    csv_obj_name = get_random_string()
+    bucket_name = "test"
+    upload_csv_object(bucket_name,csv_obj_name,csv_obj)
+
+    obj_size = len(csv_obj.encode('utf-8'))
+
+    res_s3select_response = run_s3select(bucket_name,csv_obj_name,"select sum(int(_1)) from s3object;",progress = True)
+    records_payload_size = len(remove_xml_tags_from_result(  run_s3select(bucket_name,csv_obj_name, 'select sum(int(_1)) from s3object;')).replace("\n",""))
+
+    progress_no = math.ceil(obj_size/4194304)
+    
+    # To do: Validate bytes processed after supporting compressed data
+    n = 1
+    for i in range(0, progress_no-1):
+        s3select_assert_result(res_s3select_response[i]['Details']['BytesScanned'], 4194304*n)
```

Comment:
not sure it will give the same result in the different environments (test cluster)
theses chunks may have different sizes (which will cause the test to fail).
but all chunks should sum up to object size.

Reply:
Yeah, this test was on the assumption that size of each chunk except the last one is a constant.

Reply:
:+1: